src/components/PopUpWithConfirm.js

```diff
@@ -0,0 +1,39 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector, confirmHandler) {
+    super(popupSelector);
+    this._confirmHandler = confirmHandler;
+  }
+
+  open() {
+    super.open();
+  }
+
+  close() {
+    super.close();
+  }
+
+  submit = (evt) => {
+    evt.preventDefault();
+    this._handleConfirm();
+
+    this.close();
```

Comment:
Needs correction: you should close the popup only after the successful api response in index.js